src/wallet/wallet_rpc_server_commands_defs.h

```diff
@@ -2679,5 +2679,69 @@ namespace wallet_rpc
     typedef epee::misc_utils::struct_init<response_t> response;
   };
 
+  struct COMMAND_RPC_SETUP_BACKGROUND_SYNC
+  {
+    struct request_t
+    {
+      std::string background_sync_type;
+      std::string wallet_password;
+      std::string background_cache_password;
+
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(background_sync_type)
+        KV_SERIALIZE(wallet_password)
+        KV_SERIALIZE_OPT(background_cache_password, (std::string)"")
+      END_KV_SERIALIZE_MAP()
+    };
+    typedef epee::misc_utils::struct_init<request_t> request;
+
+    struct response_t
+    {
+      BEGIN_KV_SERIALIZE_MAP()
+      END_KV_SERIALIZE_MAP()
+    };
+    typedef epee::misc_utils::struct_init<response_t> response;
+  };
+
+  struct COMMAND_RPC_START_BACKGROUND_SYNC
+  {
+    struct request_t
+    {
+      BEGIN_KV_SERIALIZE_MAP()
+      END_KV_SERIALIZE_MAP()
+    };
+    typedef epee::misc_utils::struct_init<request_t> request;
+
+    struct response_t
+    {
+      BEGIN_KV_SERIALIZE_MAP()
+      END_KV_SERIALIZE_MAP()
+    };
+    typedef epee::misc_utils::struct_init<response_t> response;
+  };
+
+  struct COMMAND_RPC_STOP_BACKGROUND_SYNC
+  {
+    struct request_t
+    {
+      std::string wallet_password;
+      std::string seed;
+      std::string seed_offset;
```

Comment:
I don't know much about security and am not as familiar with the codebase as you, so forgive me if this question comes off as naive, but sending the `seed` and `wallet_password` seems suspect. Perhaps I'm confusing `seed` here with what we traditionally think of when we think of a wallet seed (the private spend key). If this is not the actual private spend key / seed used to access the users' funds, consider renaming it, so that it doesn't confuse less-familiar users like myself.

Reply:
It's the wallet seed, following the pattern from `COMMAND_RPC_RESTORE_DETERMINISTIC_WALLET`.

Starting background sync wipes the spend key from memory, so stopping background sync needs to bring the spend key back in memory.

A user would need to pass the wallet password here if the user's seed is stored in an encrypted file on disk. The password is used to decrypt the file contents and reload the spend key back in memory to process the background cache.

A user would need to pass the seed here if the user's seed isn't stored on disk (if the wallet is an in-memory wallet as is the case in some of the functional tests).

I'm open to an alternative approach to passing strings here if you have one in mind. @moneromooo-monero had one [here](https://github.com/monero-project/monero/pull/8619#issuecomment-1632951461) but that seemed particularly heavy and out of scope for this PR. Another could be leaning on http auth (and taking greater care with auth material server-side), which also seems heavy for this PR.

Reply:
Thanks for explaining. This makes more sense to me now. For my own understanding, and after reading over Moneromooo's comment, it seems like with the current setup it would be on-the-user to ensure that the plaintext (json) traffic to and from the RPC wallet is encrypted, is this correct (assuming the other suggested avenues are outside the scope of this PR)?

Reply:
The wallet RPC [supports SSL](https://monerodocs.org/interacting/monero-wallet-rpc-reference/) or it can be exposed behind a hidden service.

Perhaps RPC endpoints that expose spend key material in the request/response bodies could require an SSL connection by default, and allow the setting to be overwritten at startup via a flag. I also think that would be out of scope for this PR considering all the endpoints that already support spend key material in request params. But I catch your drift in calling this out. It's a good thing to point out in review.